src/core/VLIW/VLIW.ts

```diff
@@ -166,6 +167,8 @@ export class VLIW extends Machine {
 
                 if (!this.functionalUnit[type][index].isFree() ||
                     DependencyChecker.checkNat(instruction.getOperation(i), this._NaTGP, this._NaTFP)) {
+                    //TODO: This really fails when there is a RAW dependency?
+                    return VLIWError.ERRRAW; // VLIW_ERRRA;
```

Comment:
We should also refactor VLIW errors into exceptions too. In my opinion, returning error codes might be interesting for execution stats, but it's definitely not for actual control flow
EDIT: I mean, in a TypeScript app, error codes are a totally valid thing in low-level software development!

Regarding the TODO above: I'm almost sure this should not be considered a RAW hazard for the entire logic of the conditional clause

Reply:
VLIWError class is strange, it is not only used as an error class but also as an VLIW machine actual status. While I agree that some statuses should not stop the execution, there are some that definitely needs to stop the machine. Also, in this tic function, there is an boolean variable stopFlow that in theory should stop the machine, but it does not work as intended in some cases. 

Regarding this return, maybe is inecessary but as there were an `stopFlow = true`, which was not working and breaking the actual execution, I thought the original intention was to stop the machine when an dependency was found ant show a message but was left semi unimplemented.

Finally, about the TODO, I tested with a code which had an RAW hazard and, in fact, this code was triggered. idk is this was the intended purpose.